paper-api/src/main/java/io/papermc/paper/world/damagesource/CombatTracker.java

```diff
@@ -0,0 +1,104 @@
+package io.papermc.paper.world.damagesource;
+
+import net.kyori.adventure.text.Component;
+import org.bukkit.entity.LivingEntity;
+import org.jetbrains.annotations.ApiStatus;
+import org.jspecify.annotations.NullMarked;
+import org.jspecify.annotations.Nullable;
+
+import java.util.List;
+
+/**
+ * Represents entity's combat tracker
+ *
+ * @since 1.21.4
+ */
+@NullMarked
+@ApiStatus.Experimental
+@ApiStatus.NonExtendable
+public interface CombatTracker {
+
+    /**
+     * Gets the entity behind this combat tracker
+     *
+     * @return the entity behind this combat tracker
+     */
+    LivingEntity getEntity();
+
+    /**
+     * Gets the list of recorded combat entries.
+     * <br>
+     * Mutating the list is safe and won't have
+     * an effect on this entity's combat history.
```

Comment:
It should just be immutable/say it's a copy

Reply:
Updated the wording to state it's a copy.
Haven't made this immutable because it's returning a copy internally each time, so modifications are safe. Otherwise, users would need to make a copy of a copy.